src/hotspot/share/compiler/tieredThresholdPolicy.cpp

```diff
@@ -826,6 +826,18 @@ void TieredThresholdPolicy::create_mdo(const methodHandle& mh, Thread* THREAD) {
   if (mh->method_data() == NULL) {
     Method::build_interpreter_method_data(mh, CHECK_AND_CLEAR);
   }
+  if (ProfileInterpreter) {
+    MethodData* mdo = mh->method_data();
+    if (mdo != NULL) {
+      JavaThread* jt = THREAD->as_Java_thread();
+      frame last_frame = jt->last_frame();
+      if (last_frame.is_interpreted_frame()) {
+        int bci = last_frame.interpreter_frame_bci();
```

Comment:
Please, add assert or guarantee to check that `last_frame.interpreter_frame_method()` is the same as `mh.

Reply:
Good point. I added an explicit check to make this more robust.